### ISSUE TITLE
Make it possible to get width and height from imageUrlBuilder

### ISSUE THREAD

**Issue by Svish** [sanity-io/sanity#1627]
**Is your feature request related to a problem? Please describe.**
When for example lazy loading and/or animating images on a website, you often need to specify the width _and_ height of the image.

This is currently hard/impossible to do when using the `imageUrlBuilder` from `@sanity/image-url` because you don't know what actual dimensions the image will have when it comes back after resizing and cropping.

For example in this case, I would only know the width, and not the resulting height:

```html
          <img
            key={project.image.asset._ref}
            src={imageUrlBuilder(config).image(project.image)
              .width(400)
              .format('jpg')
              .auto('format')
              .quality(80)
              .url()}
          />
```

You can get the _original_ width and height from the asset ref, but that does not include any cropping or focal points or anything along those lines.

**Describe the solution you'd like**
I'd like a an alternative to `.url()` named `.props()` which returned an object with `src`, `height`, and `width`, where height and width was calculated the exact same way as the Sanity image API does, meaning the height and width would match the exact height and width of the image returned from the url. Returning an object like that means you could both access the height and width as you want, but also that you could easily spread it directly onto an img tag:
          <img
            key={project.image.asset._ref}
            {...imageUrlBuilder(config).image(project.image)
              .width(400)
              .format('jpg')
              .auto('format')
              .quality(80)
              .props()}
          />

**Describe alternatives you've considered**
Calculating the resulting height and width manually would be both hard and error-prone. Don't even know how I'd begin to do that with all the hotspot, crop, and size parameters... 😕

**Comment by kmelve** [comment]
This is an interesting suggestion. We'll look into it when we iterate on the image-url builder!

**Comment by nullhook** [comment]
Bump.

Would love to have this feature too.

**Comment by noii98** [comment]
function croppedDimensions(height, width, crop, croppedWidth){
  if(crop){
    const y = crop.top + crop.bottom
    const x = crop.left + crop.right
    height = (1-y) * height
    width = Math.round((1-x) * width)
  }
  height = croppedWidth ? Math.round(height * croppedWidth / width) : Math.round(height)
  return {height, width: croppedWidth ?? width}
}

You can use this function to get the height and width.

**Comment by bstrong-elitedigital** [comment]
Any updates on this?

**Comment by studio-jw** [comment]
Bump!

**Comment by raulfdm** [comment]
Plus one for that.

I thought we would have a way of getting all image information like we can do via GROQ but we can't. :/

It would be nice to have a method like: `image.info()` or `image.meta()` and get everything we get when we do:

```
"image": image.asset ->
```

**Comment by pm0u** [comment]
+1 here as well, or at least a well explained "proper" way to do this... it's best practice now to have w/h on images and when it's just an asset reference we are stuck making another API call just for this information

**Comment by coreyward** [comment]
@pm0u For what it is worth, if you're using CSS to size your images, the `width` and `height` properties only need to convey the aspect ratio to the browser. Unless you're cropping the image, using the original width and height supplied via the image `_id` or `_ref` would work for this purpose.

**Comment by Svish** [comment]
@coreyward Being able to crop the images, with selected focus points and such, is part of the "point" with sanity images though.

It should be a proper and stable way to get the original height, width, and aspect ratio, as well as _cropped_ height, width, and aspect ratio, without hacks.

**Comment by pm0u** [comment]
@coreyward I looked into how `next-sanity-image` was accomplishing this and they use those dimensions, I am building a similar helper to work with picture elements. I am planning to use those for now, although I have to say I'm never excited about using a filename as a source of truth as there can be lots of things that I could see breaking this functionality. Perhaps those concerns are moot if the naming conventions of sanity are static, but I wouldn't necessarily expect that to be guaranteed. If it's already there in the filename, it seems like it should be easy enough to expose these values in a future proof way.

**Comment by coreyward** [comment]
@pm0u The Sanity docs do say that this ID format is stable and consistent, and in fact use this information in the Studio and elsewhere similarly.

**Comment by coreyward** [comment]
@Svish Agreed! It would make my work on gatsby-plugin-sanity-image much easier as well!

**Comment by pm0u** [comment]
@coreyward thanks for pointing that out, I finally found that mentioned here: https://www.sanity.io/docs/image-urls which I hadn't read before. Wish it was a bit more prominent but good to know!

**Comment by coreyward** [comment]
@nikolailehbrink That returns the dimensions of the uploaded image, not the output image (post-crop, post-transformations). 

If anybody wants to determine the output dimensions deterministically, take a look at [sanity-image](https://github.com/coreyward/sanity-image). It does not use this library to build URLs, but instead builds the URL from your parameters using its own approach in a way that allows for predicting reliably the output dimensions across all cases. You can use it as either a React component or as a function that returns the dimensions and a full URL to the image, or as a function that returns an entire src-set for you.

**Comment by nikolailehbrink** [comment]
Hi @coreyward! Thanks for the correction. Tried it out and know now what the actual issue here is about!
